### PR TITLE
Fix speech_llm recipe

### PR DESCRIPTION
**Pull request by yfyeung**
- [x] fix `model.eval()` and `model.train()` in `train,py`, resulting in consistent performance improvement
- [x] fix some minor bugs in `train.py` and `decode.py`
- [x] use `compute_CER=True` instead of reimplementation in `decode.py`
- [x] add missing backslash in `RESULT.md`
- [x] remove ineffective `normalize_text_alimeeting`; no effect on AISHELL. Suggest applying normalization during preprocessing for multi-dataset
- [x] remove `filter_uneven_sized_batch`

### CODE REVIEW

**Comment by yfyeung** [comment]
Origin:
Epoch | Avg | Beam Size | CER | Epoch | Avg | Beam Size | CER
-- | -- | -- | -- | -- | -- | -- | --
10 | 1 | 1 | 5.01 | 10 | 1 | 1 | 4.41
10 | 2 | 1 | 4.82 | 10 | 2 | 1 | 4.08
10 | 3 | 1 | 4.65 | 10 | 3 | 1 | 3.95
10 | 4 | 1 | 4.67 | 10 | 4 | 1 | 3.81
10 | 5 | 1 | 4.59 | 10 | 5 | 1 | 3.74
10 | 6 | 1 | 4.58 | 10 | 6 | 1 | 3.63
10 | 7 | 1 | 4.58 | 10 | 7 | 1 | 3.56
10 | 8 | 1 | 4.55 | 10 | 8 | 1 | 3.49
10 | 9 | 1 | 4.58 | 10 | 9 | 1 | 3.44

Now:
Epoch | Avg | Beam Size | CER | Epoch | Avg | Beam Size | CER
-- | -- | -- | -- | -- | -- | -- | --
10 | 1 | 1 | 4.77 | 10 | 1 | 1 | 4.26
10 | 2 | 1 | 4.56 | 10 | 2 | 1 | 3.95
10 | 3 | 1 | 4.49 | 10 | 3 | 1 | 3.8
10 | 4 | 1 | 4.44 | 10 | 4 | 1 | 3.69
10 | 5 | 1 | 4.44 | 10 | 5 | 1 | 3.61
10 | 6 | 1 | 4.41 | 10 | 6 | 1 | 3.57
10 | 7 | 1 | 4.59 | 10 | 7 | 1 | 3.47
10 | 8 | 1 | 4.55 | 10 | 8 | 1 | 3.45
10 | 9 | 1 | 4.6 | 10 | 9 | 1 | 3.44

**Review by yuekaizhang — APPROVED** [review]
Many thanks!

**Comment by JinZr** [comment]
we can merge this one first and have the flake8 issue fixed in a separate pull request
